File: 1_class_object.cpp

#include <iostream>
#include <string>
using namespace std;

class Person {  // class name
public:
    // Attributes
    string name;
    string address;

    // Constructor
    Person(string name, string address) {
        this->name = name;         // Assigning the name attribute
        this->address = address;   // Assigning the address attribute
        cout << name << " " << address << endl;  // Printing name and address in constructor
    }

    // Method to introduce the person
    string intro() {
        return "My name is " + name + " and I live in " + address;
    }
};

int main() {
    // Creating objects (instances) of the Person class
    Person p1("Foysal Hossain Munna", "Dhanmondi");
    Person p2("Rakib Sarker", "Elephant Road");
    Person p3("Tasfia", "New Market");

    // Accessing the intro method of the last object (p3)
    cout << p3.intro() << endl;

    return 0;
}






























// #include<bits/stdc++.h>
// using namespace std;

// // class A{
// //     public:
// //     int value;
// //     //constructor
// //     A(int value){
// //         this->value = value;
// //         cout<<"Constructor Called"<<endl;
// //     }

// //     //method
// //     void display(){
// //         cout<<"Hello\n";
// //     }

// //     //destructor
// //     ~A(){
// //         cout<<"Desonstructor Called"<<endl;
// //     }
// // };
                    
// // int main(){
// //     A obj(8);
// //     obj.display();
// //     // obj.value = 5;
// //     cout<<obj.value<<endl;
            
              
// //     return 0;
// // }








// class A{
// public:
//     A(){
//         printf("I am Constructor\n");
//     }

//     void display(){
//         cout<<"I am Display Method\n";
//     }

//     ~A(){
//         cout<<"I am Destructor\n";
//     }

// };

// int main(){
//     A obj;
//     obj.display();

// }

File: 1_class_object.py
class Person:  #class name 
    
    species = "Homo sapiens"  # Class attribute (shared by all instances)
    def __init__(self , name , address) -> None: #constructor 
        
        self.name = name     #attribute
        self.address = address   #attribute
        print(self.name , self.address )
        
    def intro(self):  #methods
        return (f"My name is {self.name} and I live in {self.address}")



        # Class method
    @classmethod
    def get_species(cls):
        return f"All humans are {cls.species}"

p = Person('Foysal Hossain MUnna' , 'Dhonmondi')  # object or instance 
p = Person('Rakib sarker' , 'Elephant Road')  # object or instance 
p = Person('Tasfia' , 'new Market')  # object or instance 
print(p.get_species())
print(p.intro())


    





























# *************************************************

class A:
    value = 5 #class variable
    def __init__(self) -> None:
        print("I am Constructor")
    
    def display(self):
        print("I am method")

    def __del__(self):
        print("I am destructor")

    def __str__(self) -> str:
        return "I am ClasS A"




obj = A()
obj1 = A()
obj.display()
obj.value = 5

print(obj.__dict__)





print(obj.value)
print(A.value)






